modules/core/src/main/java/io/fluo/core/impl/SharedResources.java

```diff
@@ -79,6 +81,14 @@ public CuratorFramework getCurator() {
     return curator;
   }
   
+  public synchronized OracleClient getOracleClient() {
+    checkIfClosed();
+    if (oracleClient == null) {
+      oracleClient = new OracleClient(env);
+    }
+    return oracleClient;
```

Comment:
Now that shared resources is tracking it, we should probably close it.  I was looking at the code for OracleClient and I think I know what needs to be done to add a close method (also I have worked on this code w/ @cjnolet  before).  I am thinking of making pull request against your branch to add that.  Is that ok or would you rather work on adding the close method?